rust/altrios-core/src/train/speed_limit_train_sim.rs

```diff
@@ -435,16 +461,23 @@ impl SpeedLimitTrainSim {
             // ```
             self.state.pwr_whl_out + self.loco_con.state.pwr_rate_out_max * self.state.dt,
         ));
+
+        //calculate the max braking that a consist can apply.  Same as set speed train sim.
         let pwr_neg_max = self.loco_con.state.pwr_dyn_brake_max.max(si::Power::ZERO);
         ensure!(
             pwr_pos_max >= si::Power::ZERO,
             format_dbg!(pwr_pos_max >= si::Power::ZERO)
         );
+
+        //need for energy calc below.  This is structured differently than set_speed_train_sim.
+        //However, I think the math is the same maybe.
+        //Would it be worth the effort to calculate this once when we initialize and save it as a parameter of the train?  Then we would not have to calculate it every dt.
```

Comment:
Probably not because it's a cheap calculation, and since I have aspirations of eventually incorporating our [rust solver package](https://crates.io/crates/dess) to optionally have variable time step solvers, I would not want to had code this.